federator/src/lib/FederatorERC.ts

```diff
@@ -0,0 +1,614 @@
+    logs,
+    currentBlock,
+    mediumAndSmall,
+    confirmations,
+    sideChainConfig,
+    federationFactory,
+    allowTokensFactory,
+    bridgeFactory,
+  }: {
+    sideChainId: number;
+    mainChainId: number;
+    transactionSender: TransactionSender;
+    logs: any;
+    currentBlock: number;
+    mediumAndSmall: boolean;
+    confirmations: { mediumAmountConfirmations: number; largeAmountConfirmations: number };
+    sideChainConfig: ConfigChain;
+    federationFactory: FederationFactory;
+    allowTokensFactory: AllowTokensFactory;
+    bridgeFactory: BridgeFactory;
+  }) {
+    try {
+      const federatorAddress = await transactionSender.getAddress(this.config.privateKey);
+      const sideFedContract = await federationFactory.getSideFederationContract();
+      const allowTokens = await allowTokensFactory.getMainAllowTokensContract();
+
+      await this.checkFederatorIsMember(sideFedContract, federatorAddress);
+
+      for (const log of logs) {
+        await this.processLog({
+          sideChainId,
+          mainChainId,
+          transactionSender,
+          log,
+          confirmations,
+          sideFedContract,
+          allowTokens,
+          currentBlock,
+          federatorAddress,
+          mediumAndSmall,
+          sideChainConfig,
+          allowTokensFactory,
+          bridgeFactory,
+        });
+      }
+
+      return true;
+    } catch (err) {
+      throw new CustomError(`Exception processing logs`, err);
+    }
+  }
+
+  async _voteTransaction({
+    sideChainId,
+    mainChainId,
+    transactionSender,
+    sideFedContract,
+    tokenAddress,
+    sender,
+    receiver,
+    amount,
+    symbol,
+    blockHash,
+    transactionHash,
+    logIndex,
+    decimals,
+    granularity,
+    typeId,
+    txId,
+    federatorAddress,
+    originChainId,
+    destinationChainId,
+    sideChainConfig,
+  }: {
+    sideChainId: number;
+    mainChainId: number;
+    transactionSender: TransactionSender;
+    sideFedContract: IFederationV3 | IFederationV2;
+    tokenAddress: string;
+    sender: string;
+    receiver: string;
+    amount: BN;
+    symbol: string;
+    blockHash: string;
+    transactionHash: string;
+    logIndex: number;
+    decimals: string;
+    granularity: string;
+    typeId: number;
+    txId: string;
+    federatorAddress: string;
+    originChainId: number;
+    destinationChainId: number;
+    sideChainConfig: ConfigChain;
+  }) {
+    try {
+      txId = txId.toLowerCase();
+      this.logger.info(
+        `TransactionId ${txId} Voting Transfer ${amount} of originalTokenAddress:${tokenAddress} trough sidechain bridge ${sideChainConfig.bridge} to receiver ${receiver}`,
+      );
+
+      const txDataAbi = await sideFedContract.getVoteTransactionABI({
+        originalTokenAddress: tokenAddress,
+        sender,
+        receiver,
+        amount,
+        blockHash,
+        transactionHash,
+        logIndex,
+        tokenType: utils.tokenType.COIN,
+        originChainId,
```

Comment:
originChainId can be infered by the federation contract?

Reply:
actually is coming from the event log object 
![image](https://user-images.githubusercontent.com/17556614/144895314-7dcef552-e318-4448-81ab-9aec34dedff4.png)